Add WireGuard PrivateKey/PresharedKey rule

gemesa opened the pull request:
This is a rule I regularly use to look for WireGuard .conf files (specifically private and preshared WG keys). Maybe it makes sense to add it to the defaults. I generated the 2 keys in the example myself just for this purpose.

bradlarsen commented on crates/noseyparker/data/default/builtin/rules/wireguard.yml:
```diff
@@ -0,0 +1,28 @@
+  pattern: |
+    (?x)
+    PrivateKey|PresharedKey\s*=\s*
+    ([A-Za-z0-9+/]{43}={1})
```
```suggestion
    ([A-Za-z0-9+/]{43}=)
```

gemesa replied:
fixed

bradlarsen reviewed (changes requested):
Thank you @gemesa! This looks really good. This will allow Nosey Parker to detect more secrets with precision.

Would you please split this new `WireGuard PrivateKey/PresharedKey` rule into two new rules? They are conceptually different things, even if they have the same format. `WireGuard Private Key` and `WireGuard Preshared Key` are good names for them.

bradlarsen commented on crates/noseyparker/data/default/builtin/rules/wireguard.yml:
```diff
@@ -0,0 +1,28 @@
+      Endpoint = [Server Addr:Server Port]
+
+  references:
+  - https://www.wireguard.com/quickstart/
+  - https://gist.github.com/lanceliao/5d2977f417f34dda0e3d63ac7e217fd6
```
```suggestion
  - https://www.wireguard.com/quickstart/
  - https://manpages.debian.org/testing/wireguard-tools/wg.8.en.html
  - https://gist.github.com/lanceliao/5d2977f417f34dda0e3d63ac7e217fd6
```

gemesa replied:
fixed

gemesa commented:
> Thank you @gemesa! This looks really good. This will allow Nosey Parker to detect more secrets with precision.
> 
> Would you please split this new `WireGuard PrivateKey/PresharedKey` rule into two new rules? They are conceptually different things, even if they have the same format. `WireGuard Private Key` and `WireGuard Preshared Key` are good names for them.

I split them and fixed the review findings, please check it out.

bradlarsen commented:
Thank you for your contribution, @gemesa!
